cmd/artifact-server/main.go

```diff
@@ -0,0 +1,145 @@
+	"google.cloud.apigeeregistry.v1.apihub.TaxonomyList":           &rpc.TaxonomyList{},
+	"google.cloud.apigeeregistry.v1.controller.Manifest":           &rpc.Manifest{},
+	"google.cloud.apigeeregistry.v1.controller.Receipt":            &rpc.Receipt{},
+	"google.cloud.apigeeregistry.v1.scoring.Score":                 &rpc.Score{},
+	"google.cloud.apigeeregistry.v1.scoring.ScoreCard":             &rpc.ScoreCard{},
+	"google.cloud.apigeeregistry.v1.scoring.ScoreCardDefinition":   &rpc.ScoreCardDefinition{},
+	"google.cloud.apigeeregistry.v1.scoring.ScoreDefinition":       &rpc.ScoreDefinition{},
+	"google.cloud.apigeeregistry.v1.style.ConformanceReport":       &rpc.ConformanceReport{},
+	"google.cloud.apigeeregistry.v1.style.StyleGuide":              &rpc.StyleGuide{},
+	"google.protobuf.FileDescriptorSet":                            &descriptorpb.FileDescriptorSet{},
+}
+
+func handleRequest(w http.ResponseWriter, req *http.Request) {
+	ctx := context.Background()
+	client, err := connection.NewRegistryClient(ctx)
+	if err != nil {
+		writeError(w, err, http.StatusInternalServerError)
+		return
+	}
+	name := strings.TrimPrefix(req.URL.Path, "/")
+	contents, err := client.GetArtifactContents(ctx, &rpc.GetArtifactContentsRequest{Name: name})
+	if err != nil {
+		writeError(w, err, http.StatusNotFound)
+		return
+	}
+	if contents.GetContentType() == "text/plain" {
+		fmt.Fprintf(w, "%s\n", string(contents.GetData()))
+		return
+	}
+	messageType, err := core.MessageTypeForMimeType(contents.GetContentType())
+	if err != nil {
+		writeError(w, err, http.StatusInternalServerError)
+		return
+	}
+	message := messageTypes[messageType]
+	if message == nil {
+		err = fmt.Errorf("unsupported message type: %s", messageType)
+		writeError(w, err, http.StatusInternalServerError)
+		return
+	}
+	if err := proto.Unmarshal(contents.GetData(), message); err != nil {
+		writeError(w, err, http.StatusInternalServerError)
+		return
+	}
+	accept := req.Header["Accept"]
+	if slices.Contains(accept, "text/json") {
+		fmt.Fprint(w, protojson.Format(message))
+		fmt.Fprint(w, "\n")
+		return
+	}
+	if slices.Contains(accept, "text/yaml") {
+		fmt.Fprint(w, yamlFormat(message))
+		return
+	}
+	if slices.Contains(accept, "text/plain") {
+		fmt.Fprint(w, prototext.Format(message))
+		return
+	}
+	// default return format is JSON
+	fmt.Fprint(w, protojson.Format(message))
+	fmt.Fprint(w, "\n")
+}
+
+func yamlFormat(message proto.Message) string {
+	var node yaml.Node
+	err := yaml.Unmarshal([]byte(protojson.Format(message)), &node)
+	if err != nil {
+		return err.Error()
+	}
+	styleForYAML(&node)
+	b, err := yaml.Marshal(node.Content[0])
+	if err != nil {
+		return err.Error()
+	}
+	return string(b)
+}
+
+func styleForYAML(node *yaml.Node) {
+	node.Style = 0
+	for _, n := range node.Content {
+		styleForYAML(n)
+	}
+}
+
+func writeError(w http.ResponseWriter, err error, code int) {
+	http.Error(w, err.Error(), code)
+}
```

Comment:
I'm not really seeing the benefit to this function. Why not call `http.Error` directly as needed?